conv/decode.go

```diff
@@ -71,8 +77,13 @@ func ToString(s string) (string, error) {
 	return s, nil
 }
 
+// toBytes converts a string to a byte slice with zero allocation.
+func toBytes(s string) []byte {
+	return unsafe.Slice(unsafe.StringData(s), len(s))
+}
+
 func ToBytes(s string) ([]byte, error) {
-	return []byte(s), nil
+	return toBytes(s), nil
```

Comment:
I took the liberty of changing this so it doesn't allocate memory.

Reply:
But this is only used to convert URL parameters, isn't it? Those strings are only referenced briefly and then not used again as far as I can see.

Reply:
Those strings reference values of the map returned by `URL.Query()`. In turn, the keys and values returned by `Query()` would likely reference the raw URL inside `url.URL`. This raw URL may then be referenced by any middleware or by net/http itself.

I think it's quite possible it could corrupt something unexpectedly.